packages/devtools_server/lib/src/server.dart

```diff
@@ -208,12 +216,27 @@ Future<HttpServer> serveDevTools({
           _normalizeVmServiceUri(serviceProtocolUri).toString();
     }
 
-    String url = devToolsUrl.toString();
-    // If serviceProtocolUri != null, add it to the url.
-    if (serviceProtocolUri != null) {
-      url = Uri.parse(devToolsUrl).replace(queryParameters: {
-        'uri': serviceProtocolUri,
-      }).toString();
+    final queryParameters = {
+      if (serviceProtocolUri != null) 'uri': serviceProtocolUri,
+      if (appSizeBase != null) 'appSizeBase': appSizeBase,
+      if (appSizeTest != null) 'appSizeTest': appSizeTest,
+    };
+    String url = Uri.parse(devToolsUrl)
+        .replace(queryParameters: queryParameters)
+        .toString();
+
+    // If app size parameters are present, open to the standalone `appsize`
+    // page, regardless if there is a vm service uri specified. We only check
+    // for the presence of [appSizeBase] here because [appSizeTest] may or may
+    // not be specified (it should only be present for diffs). If [appSizeTest]
+    // is present without [appSizeBase], we will ignore the parameter.
+    if (appSizeBase != null) {
+      final startQueryParamIndex = url.indexOf('?');
```

Comment:
use uri parsing instead of indexof

Reply:
when I tried to add `/#/appsize` using `Uri.replace(path: '/#/appsize)`, the `#` is encoded and the url does load properly.

Reply:
I think you want to use fragment to change the fragment property rather than path.
https://api.dart.dev/stable/2.10.4/dart-core/Uri/fragment.html

Reply:
this also doesn't work because the fragment is _after_ the query parameters (`http://127.0.0.1:9102?appSizeBase=old.json&appSizeTest=new.json#appsize`), and devtools expects the fragment id to be _before_ the query parameters (`http://127.0.0.1:9102/#/appsize?appSizeBase=old.json&appSizeTest=new.json`)

Reply:
where in devtools are we requiring the fragment to be before the query parameters? that seems like a bug.

Reply:
see https://github.com/flutter/devtools/issues/2475 for some input from danny